### PR TITLE
Ensure selective masking test gracefully cleans up 

### PR DESCRIPTION
**Pull request by nyalldawson**
Otherwise it leaves files in the test report folder which subsequent github action steps cannot access, breaking the workflows

### CODE REVIEW

**Review by troopa81 — APPROVED** [review]

File: python/testing/__init__.py [hunk 1]
```diff
@@ -127,6 +130,24 @@ def write_local_html_report(cls, report: str):
         if not QgisTestCase.is_ci_run():
             QDesktopServices.openUrl(QUrl.fromLocalFile(report_file))
 
+    @classmethod
```
Comment (troopa81) [inline python/testing/__init__.py:133]: Shall this part not be in another PR ?

**Review by troopa81 — APPROVED** [review]

File: python/testing/__init__.py (hunk 1 above)
Comment (nyalldawson) [reply python/testing/__init__.py:133]: Possibly, but let's avoid unnecessary load on the ci
